controllers/vaultdynamicsecret_controller.go

```diff
@@ -710,28 +684,23 @@ func (r *VaultDynamicSecretReconciler) vaultClientCallback(ctx context.Context,
 		return
 	}
 
-	reqs := map[SyncRequest]empty{}
+	reqs := map[client.ObjectKey]empty{}
 	for _, o := range l.Items {
 		if o.Status.VaultClientMeta.CacheKey == cacheKey.String() {
-			_, delay := computeMaxJitterDuration(vdsMaxClientCallbackDelayForJitter)
-			req := SyncRequest{
-				Delay:        delay,
-				RequeueOnErr: true,
-				Request: reconcile.Request{
-					NamespacedName: types.NamespacedName{
-						Namespace: o.GetNamespace(),
-						Name:      o.GetName(),
-					},
-				},
+			evt := event.GenericEvent{
+				Object: o.DeepCopy(),
```

Comment:
IIRC events really only need the namespace and name of the object, so we might be able to avoid the deepcopy something like this:

```suggestion
				Object: &secretsv1beta1.VaultDynamicSecret{
					ObjectMeta: metav1.ObjectMeta{
						Namespace: o.Namespace,
						Name:      o.Name,
					},
				},
```

Reply:
Fixed in 3ee8cfac7d397fe70f2f1d263ce5b8e62317c33e